File: api/Models/ProductItem.cs
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    public class ProductItem
    {
        public int ProductItemId { get; set; }
        [Required]
        public double Price { get; set; }
        public int? Size { get; set; }   
        public int? PizzaType { get; set; }
        //Relation
        public int ProductId { get; set; }
        public Product Product { get; set; }
        public List<CartItem> CartItems { get; set; } = new List<CartItem>();

    }
}

File: frontend/src/Shared/Models/ProductItem.ts
import {  ProductShortGet } from "./Product"

export type ProductItemGet = {
    productItemId:number,
    price:number,
    size?:number,
    pizzaType?:number,
    productId:number,
    product:ProductShortGet
}

export type CreateCartItem = {
    productItemId:number;
    ingredientsId:number[];
}